pkg/skaffold/runner/dev.go

```diff
@@ -85,9 +86,13 @@ func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer) error {
 			r.changeSet.ResetSync()
 			r.intents.ResetSync()
 		}()
+
+		// todo: make this configurable
```

Comment:
Is this something you left here while you're coding and wanted to add before committing your changes ?

Reply:
No, not in this PR, I left it in case if anyone wants to do it, maybe I'll do it by myself, but not now

Reply:
I'd suggest to implement configurable re-try for sync and default it to `0` -- not re-try to keep the user experience the same as before, if we want to have a sync retry, but I am still not convinced that we need this. 

While `// todo` comments might seem like helpful reminders, they often create more problems than they solve:

`Limited Visibility`: Only those who stumble upon the specific code section see the comment, potentially leaving others unaware of the task.
`Unclear Ownership and Status`: There's no indication of who's responsible for the task, whether it's assigned, in progress, or forgotten.
`Lack of Discussion`: Meaningful conversations are difficult without a dedicated issue or pull request, hindering collaboration.
`Prioritization Challenges`: Maintainers struggle to assess the urgency and importance of tasks scattered throughout the codebase.
`Ambiguity in Pull Requests`: It's unclear if // todo marks unfinished work or instructions for future changes, causing confusion during reviews.
These issues often lead to // todo items being overlooked or forgotten.

A more effective approach is to create an issue for each task:

- Issues provide a centralized location for tracking, assigning, and discussing tasks.
- They allow for clear prioritization and status updates.
- Discussions within issues create better communication and collaboration.